File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        int sum = (n * (n + 1)) / 2;
        int s = 0;
        for (const auto &num: nums) {
            s += num;
        }

        return sum - s;
    }
};

File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        boolean[] checks = new boolean[nums.length + 1];

        for (int num: nums) {
            checks[num] = true;
        }

        for (int i = 0; i < checks.length; i++) {
            if (!checks[i]) {
                return i;
            }
        }

        return -1;
    }
}